openmdao/drivers/scipy_optimizer.py

```diff
@@ -449,20 +448,27 @@ def run(self):
                               f"than min allowed ({info['min_improve_pct']:.1f}%)."
                         issue_warning(msg, prefix=self.msginfo, category=DerivativesWarning)
 
+        if MPI:
+            comm = MPI.COMM_WORLD
+        else:
+            comm = FakeComm()
         # optimize
         try:
             if opt in _optimizers:
-                result = minimize(self._objfunc, x_init,
-                                  # args=(),
-                                  method=opt,
-                                  jac=jac,
-                                  hess=hess,
-                                  # hessp=None,
-                                  bounds=bounds,
-                                  constraints=constraints,
-                                  tol=self.options['tol'],
-                                  # callback=None,
-                                  options=self.opt_settings)
+                if comm.rank == 0:
```

Comment:
This doesn't seem correct since you'd only be doing the optimization on one rank and I'd expect it to hang when collective MPI calls are made in the model.